misc/testerina/modules/testerina-core/src/main/ballerina/assert.bal

```diff
@@ -69,9 +69,8 @@ public isolated function assertFalse(boolean condition, string msg = "Assertion
 # + actual - Actual value
 # + expected - Expected value
 # + msg - Assertion error message
-public isolated function assertEquals(anydata|error actual, anydata|error expected, string msg = "Assertion Failed!") {
-    boolean isEqual = (actual == expected);
-    if (!isEqual) {
+public isolated function assertEquals(anydata|error actual, anydata expected, string msg = "Assertion Failed!") {
+    if (actual is error || actual != expected) {
```

Comment:
If we don't have one already, shall we add a test for when this function is called with an `error` value as the arg for `actual`?

Reply:
Issue created https://github.com/ballerina-platform/ballerina-lang/issues/28818